source/Platforms.hx

```diff
@@ -0,0 +1,178 @@
+                velocity.x = -platformVelocity;
+            }
+        }
+        if (RIGHT) {
+            if (x >= startX + maxX) {
+               RIGHT = false;
+               LEFT = true;
+            }
+            else {
+                velocity.x = platformVelocity;
+            }
+        }
+    }
+
+    // Add or update sprites colliding with platform
+    public function platformObjects(elapsed:Float, player:FlxObject, platform:Platforms):Void
+    {
+        // If there is nothing in the touching platform group, add this object
+        if (countTouching == 0)
+        {  
+            var obj = new PlatformTracker(player, (player.x - platform.x));
+            touchingSprites.add(obj);
+            countTouching++;
+        }
+        else
+        {
+            inContact = false;
+            // Iterate through the members of the group
+            for (i in 0...countTouching)
+            {
+                // If this sprite touching the platform has already been touching the platform
+                if (touchingSprites.members[i].returnBase() == player)
+                {
+                    // Mark that this sprite has been found and was already in contact
+                    inContact = true;
+
+                    // Multiply velocity by elapsed to get the player's movement each frame.
+                    touchingSprites.members[i].trackedOffsetX += player.velocity.x * elapsed;
+
+                    // Update player x position
+                    player.x = platform.x + touchingSprites.members[i].returnOffset();
+                }
+            }
+            // If the end of the group is reached and this object isn't already in contact, add it
+            if (!inContact)
+            {
+                var obj = new PlatformTracker(player, (player.x - platform.x));
+                touchingSprites.add(obj);
+                countTouching++;
+            }
+        }
+    }
+
+    // Remove sprites NOT colliding with platform
+    private function notPlatformObjects(elapsed:Float, sprite:FlxObject, platform:Platforms):Void
+    {
+        // If there are objects touching the platform, make sure it isn't this one
+        if (countTouching != 0)
+        {
+            for (i in 0...countTouching)
+            {
+                // If this object is STILl in the list of sprites touching the platform, remove it
+                if (touchingSprites.members[i].returnBase() == sprite)
+                { 
+                    // Exit loop and Remove from touchingSprites
+                    touchingSprites.remove(touchingSprites.members[i], true);
+                    touchingSprites.update(elapsed);
+                    countTouching--;
+                }
+            }
+        }
+    }
+	
+
+
+    public function platformUpdate(elapsed:Float, objects:FlxTypedGroup<FlxObject>, platform:Platforms):Void
+    {
+        for (i in 0...objects.length)
+        {
+            if (FlxG.collide(platform, objects.members[i])) {
```

Comment:
I don't think I'll implement this... I'll look at the other changes you suggested, but this would be last on my list of things to change. You're right though, in a larger game it would definitely slow down. It wouldn't scale well.